File: CS/WindowsApplication1/XtraReport1.cs
using DevExpress.DataAccess.Sql.DataApi;
using System;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace WindowsApplication1 {
    public partial class XtraReport1: DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void XtraReport1_ParametersRequestBeforeShow(object sender,DevExpress.XtraReports.Parameters.ParametersRequestEventArgs e) {
            BaseEdit customEditor = InitCustomEditor();            
            e.ParametersInformation[0].Editor = customEditor;
        }

        private BaseEdit InitCustomEditor() {
            CheckedComboBoxEdit editor = new CheckedComboBoxEdit();            
            parameterDataSource.Fill();
            IResultSet result = parameterDataSource.Result;
            ITable categoriesTable = result["Categories"];
            foreach(IRow item in categoriesTable) {
                    editor.Properties.Items.Add(item["CategoryID"], item["CategoryName"].ToString(), CheckState.Unchecked, true);
            }
            return editor;
        }

        private void XtraReport1_ParametersRequestSubmit(object sender,DevExpress.XtraReports.Parameters.ParametersRequestEventArgs e) {
            this.FilterString = @"[CategoryID] In ("+e.ParametersInformation[0].Parameter.Value +")";
        }

    }
}

File: VB/WindowsApplication1/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors
Imports DevExpress.DataAccess.Sql.DataApi
Imports System.Windows.Forms

Namespace WindowsApplication1
    Partial Public Class XtraReport1
        Inherits DevExpress.XtraReports.UI.XtraReport
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub XtraReport1_ParametersRequestBeforeShow(ByVal sender As Object, ByVal e As DevExpress.XtraReports.Parameters.ParametersRequestEventArgs) Handles MyBase.ParametersRequestBeforeShow
            Dim editor As New CheckedComboBoxEdit()
            e.ParametersInformation(0).Editor = editor
            parameterDataSource.Fill()
            Dim result As IResultSet = parameterDataSource.Result
            Dim categoriesTable As ITable = result("Categories")

            For Each item As IRow In categoriesTable
                editor.Properties.Items.Add(item("CategoryID"), item("CategoryName").ToString(), CheckState.Unchecked, True)
            Next
        End Sub

        Private Sub XtraReport1_ParametersRequestSubmit(ByVal sender As Object, ByVal e As DevExpress.XtraReports.Parameters.ParametersRequestEventArgs) Handles MyBase.ParametersRequestSubmit
            Me.FilterString = "[CategoryID] In (" & e.ParametersInformation(0).Parameter.Value.ToString() & ")"
        End Sub

    End Class
End Namespace
